src/agent/onefuzz-supervisor/src/heartbeat.rs

```diff
@@ -54,3 +54,32 @@ pub async fn init_agent_heartbeat(queue_url: Url) -> Result<AgentHeartbeatClient
     );
     Ok(hb)
 }
+
+pub trait HeartbeatSender {
+    fn send(&self, data: HeartbeatData) -> Result<()>;
+
+    fn alive(&self) {
+        self.send(HeartbeatData::MachineAlive).unwrap()
+    }
+}
+
+impl HeartbeatSender for AgentHeartbeatClient {
+    fn send(&self, data: HeartbeatData) -> Result<()> {
+        let mut messages_lock = self
+            .context
+            .pending_messages
+            .lock()
+            .map_err(|_| Error::new(ErrorKind::Other, "Unable to acquire the lock"))?;
```

Comment:
You could also just use `anyhow::format_err!("Unable to acquire heartbeat lock")`, if you just want to create a new error value with a string message.